R/jhu_data.R

```diff
@@ -68,102 +73,176 @@ load_jhu_data <- function(
+      }
+    ))
+    
+
+    adjustments <- data.frame(fips = adjustment_state_fips, dates = as.Date(adjustment_dates))
+    if (adjustment_method == "fill_na") {
+      results <- covidData::fill_na(results = results, adjustments = adjustments)
+    }
+
+    
+    if (adjustment_method == "impute_and_redistribute") {
+       if (replace_negatives == FALSE) {
+         results = covidData::replace_negatives(data = results, measure = measure)
+       }
+
+      for (i in 1:nrow(adjustments)) {
+        adjustment_location <- adjustments[i, ]$fips
+        adjustment_date <- as.Date(adjustments[i, ]$date)
+
+        # Get state, counties and national observations for an adjustment case
+        location_data <- results %>%
+          dplyr::filter(stringr::str_sub(location, start = 1, end = 2) %in% adjustment_location |
+            location == "US" | location == adjustment_location)
+
+        # For each location in data, get imputed data
+        for (fips in unique(location_data$location)) {
+          d <- location_data[location_data$location == fips, ]
+
+          seed <- 1234
+          # Get adjusted inc column
+          imputed_inc <- covidData::adjust_daily_incidence(d, adjustments[i, ], seed, measure = measure)
+
+          # Put imputed data back to results
+          results[which(results$location == fips), ]$inc <- imputed_inc
+        }
+      }
+    }
+  }
+
+
+
+
+  # TODO: if temporal_resolution == "weekly", aggregate daily incidence to weekly incidence here
   # input: data frame with daily values, output: data frame with approximately 1/7 the # of rows
   # with weekly inc values
   # group_by(location, week), and summarize(inc = sum(inc, na.rm = TRUE))
-  
+  # if weekly temporal resolution, filter to saturdays
+  # TODO: adjust data first, go to weekly resolution later
+  if (temporal_resolution == "weekly") {
+    results <- results %>%
+      dplyr::mutate(sat_date = lubridate::ceiling_date
+      (lubridate::ymd(date), unit = "week") - 1) %>%
+      dplyr::group_by(location) %>%
+      # if the last week is not complete, drop all obs in that week
+      dplyr::filter(if (max(date) < max(sat_date)) date <= max(sat_date) - 7 else TRUE) %>%
+      dplyr::ungroup() %>%
+      # delete date column
+      dplyr::select(-2) %>%
+      dplyr::rename(date = sat_date) %>%
+      dplyr::group_by(location, date) %>%
+      dplyr::summarize(inc = sum(inc, na.rm = FALSE)) %>%
+      dplyr::ungroup()
+  }
+
   # TODO: aggregate inc to get cum
-  # at this point you'll have a data frame with results for all locations
+  # at this point you"ll have a data frame with results for all locations
   # and either an incorrect cum column that we need to replace (if temporal_resolution == daily)
   # or no cum column  (since previous step got rid of it if temporal_resolution == weekly)
   # group_by(location) [but not week] and summarize(cum = cumsum(inc))
+  # add cumulative col
```

Comment:
Let's edit this comment to remove my instructions and replace with a description of what it's doing.